pkg/prometheus/promcfg_test.go

```diff
@@ -206,32 +228,32 @@ scrape_configs: []
 		}
 
 		cg := mustNewConfigGenerator(t, p)
+		t.Run(fmt.Sprintf("case %s", tc.Scenario), func(t *testing.T) {
+			cfg, err := cg.Generate(
+				p,
+				map[string]*monitoringv1.ServiceMonitor{},
+				nil,
+				nil,
+				&assets.Store{},
+				nil,
+				nil,
+				nil,
+				nil,
+			)
 
-		cfg, err := cg.Generate(
-			p,
-			map[string]*monitoringv1.ServiceMonitor{},
-			nil,
-			nil,
-			&assets.Store{},
-			nil,
-			nil,
-			nil,
-			nil,
-		)
-		if tc.ExpectedErr != nil {
-			if tc.ExpectedErr.Error() != err.Error() {
-				t.Logf("\n%s", pretty.Compare(tc.ExpectedErr.Error(), err.Error()))
-				t.Fatal("expected error and actual error do not match")
+			if err != nil && !tc.ExpectedErr {
+				t.Fatalf("expected no error, got: %v", err)
+			}
+			if err == nil && tc.ExpectedErr {
+				t.Fatalf("expected an error, got nil")
+			}
```

Comment:
If we expect an error, no need to check for the returned configuration.

```suggestion
			if tc.ExpectedErr {
				if err == nil {
					t.Fatalf("expected an error, got nil")
				}
				return
			}
```